### ISSUE TITLE
Can't see enums implementing a protocol?

### ISSUE THREAD

**Issue by bejar37** [krzysztofzablocki/Sourcery#34]
I'm trying to write a template to deserialize enums, and trying to generate code for all enums that implement a phantom protocol, `AutoEnumDecodable`. However, it seems like `types.implementing.Protocol` doesn't work for enum types?
```
{% for type in types.implementing.AutoEnumDecodable %}
extension {{type.name}}: JsonDecodable {
    static func parse(decoder: CupertinoDecoder) {
        //...
    }
}
{% endfor %}
```
Enums that implement `AutoEnumEncodable` aren't generating anything.
BTW, I'm really enjoying using this library so far!

**Comment by ilyapuchka** [comment]
Hi @bejar37. Can you provide an example of your enum?

**Comment by bejar37** [comment]
sure, just tried with this:
```swift


enum IntOrString: AutoEnumDecodable {
    case s(string: String)
    case i(int: Int)
}
```

**Comment by ilyapuchka** [comment]
There is clearly an issue with detecting enum rawValue type. Currently parser assumes that the first type after ":" is a RawType, which is wrong when you just conform to protocol. Will try to fix that.

**Comment by bejar37** [comment]
Ah, that seems reasonable! I wouldn't mind doing:
```swift
enum IntOrString {
    case s(string: String)
    case i(int: Int)
}
extension IntOrString: AutoEnumEncodable { }
```

But this doesn't work either. That's probably the same as #35 ?

**Comment by ilyapuchka** [comment]
I'm afraid even with #35 fixed we need to make some fixes for that problem too. I'm currently looking for solutions.

**Comment by krzysztofzablocki** [comment]
I think if we waited with rawType resolution until `uniqueTypes` gets called then we can resolve this because we can identify whether a type is a protocol, class, struct etc. what do you think @ilyapuchka ?

**Comment by ilyapuchka** [comment]
That's exactly what I'm doing now 😉

**Comment by jpeckner** [comment]
@ilyapuchka @krzysztofzablocki I'm still seeing this issue using 0.16.1. I have `protocol MyAutoCellType` in a framework that my app target is linked to, and `enum MyTestEnum` in my app target. `MyAutoCellType` only gets recognized by Sourcery templates if it's not the first conformance declared on a enum (or, if I temporarily move `MyAutoCellType` into my app target itself, this issue goes away, but I need to avoid that option).

For example, Sourcery sees the `MyAutoCellType` conformance here:

    enum MyTestEnum: Equatable, MyAutoCellType {
        // sourcery: cellType = "MyTableViewCell"
        case test
    }
    
and here:

    enum MyTestEnum: String, MyAutoCellType {
        // sourcery: cellType = "MyTableViewCell"
        case test
    }
    
but not here:

    enum MyTestEnum: MyAutoCellType {
        // sourcery: cellType = "MyTableViewCell"
        case test
    }
    
and not here:

    enum MyTestEnum: MyAutoCellType, Equatable {
        // sourcery: cellType = "MyTableViewCell"
        case test
    }

I verified this by adding the following dump to my template. With the first two examples, the dump has `inheritedTypes = ["MyAutoCellType"]`, but with the last two, `inheritedTypes` doesn't contain "MyAutoCellType".

    {% for type in types.enums where type.localName == "MyTestEnum" %}
    // {{ type }}
    {% endfor %}

I did find that declaring the protocol conformance on an extension (i.e. `extension MyTestEnum: MyAutoCellType`) gets Sourcery to recognize the conformance. So given that there are at least two workarounds for this problem (either putting another declaration in front of your custom protocol; or using an extension), it might not be a high-priority bug, but FYI that it's still out there.
